src/tiled/changewangsetdata.cpp

```diff
@@ -42,43 +43,107 @@ ChangeWangSetEdges::ChangeWangSetEdges(TilesetDocument *tilesetDocument,
                                        int newValue)
     : QUndoCommand(QCoreApplication::translate("Undo Commands",
                                                "Change Wang Set edge count"))
+
+    , mTilesetDocument(tilesetDocument)
     , mWangSetModel(tilesetDocument->wangSetModel())
     , mIndex(index)
     , mOldValue(tilesetDocument->tileset()->wangSet(index)->edgeColors())
     , mNewValue(newValue)
+    , mClearTiles(nullptr)
 {
+    //when edge size changes, all tiles with wangIds need to be updated.
+    if (Tileset *tileset = mTilesetDocument->tileset().data()) {
+        if (WangSet *wangSet = tileset->wangSet(index)) {
+            mAffectedTiles = wangSet->tilesWithWangId();
+
+            if (mNewValue < mOldValue) {
+                //when the size is reduced, some wang assignments can be lost.
+                const QList<Tile *> &changedTiles = wangSet->tilesChangedOnSetEdgeColors(mNewValue);
+
+                if (!changedTiles.isEmpty()) {
+                    QList <ChangeTileWangId::WangIdChange> changes;
+
+                    for (Tile *tile : changedTiles)
+                        changes.append(ChangeTileWangId::WangIdChange(wangSet->wangIdOfTile(tile), 0, tile));
+
+                    mClearTiles = new ChangeTileWangId(mTilesetDocument, wangSet, changes, this);
```

Comment:
I understand that part. I meant that it doesn't need to be referenced by a member variable.